File: src/main/java/bad/robot/radiate/ui/StatusFrames.java
package bad.robot.radiate.ui;

import bad.robot.radiate.monitor.Observer;

import java.awt.*;
import java.util.ArrayList;
import java.util.stream.Stream;

@Deprecated
class StatusFrames {

    private final java.util.List<StatusFrame> frames = new ArrayList<>();

    StatusFrames(FrameFactory frames) {
        this.frames.addAll(frames.create());
    }

    StatusFrame primary() {
        return frames.get(0);
    }

    public void display() {
        frames.forEach(StatusFrame::display);
    }

    public void dispose() {
        frames.forEach(Window::dispose);
    }

    public Stream<Observer> createStatusPanels() {
        return frames.stream().map(StatusFrame::createStatusPanel);
    }

}

File: src/main/scala/bad/robot/radiate/ui/StatusFrames.scala
package bad.robot.radiate.ui

import bad.robot.radiate.monitor.Observer

class StatusFrames(factory: FrameFactory) {

  private val frames = factory.create

  private[ui] def primary: StatusFrame = {
     frames.head
  }

  def display() {
    frames.foreach(frame => frame.display())
  }

  def dispose() {
    frames.foreach(frame => frame.dispose())
  }
  
  def createStatusPanels: List[Observer] = {
     frames.map(frame => frame.createStatusPanel)
  }
}